### ISSUE TITLE
Allows to remove assignations from AST

### ISSUE THREAD

**Issue by jecisc** [guillep/pharo-core#5997]
In Pharo AST it is possible to remove a message sent by sending #removeFromTree to it. 

It is not possible to do it on assignations. 

I think that they should be polymorphic on this point.

**Comment by jecisc** [comment]
Same for RBTemporaryNode and probably others

**Comment by MarcusDenker** [comment]
Yes, we already looked at this with Pierre a bit.. the whole deletion logic needs some thinking and then improvement (there is removeNode:, too)

**Comment by hogoww** [comment]
From what I remember from my talk with Marcus, is that it's very easy to interpret this kind of API differently, and it looks not that good as a general purpose API.
You can take a look at #5766, this is related.
